assesspy/metrics.py

```diff
@@ -0,0 +1,375 @@
+    )
+
+    return prd
+
+
+def prb(
+    estimate: Union[list[int], list[float], pd.Series],
+    sale_price: Union[list[int], list[float], pd.Series],
+) -> float:
+    r"""
+    PRB is an index of vertical equity that quantifies the
+    relationship between ratios and estimated values as a percentage. In
+    concrete terms, a PRB of 0.02 indicates that, on average, ratios increase
+    by 2\% whenever the estimated values increase by 100 percent.
+
+    PRB is centered around 0 and has a generally accepted value of between
+    -0.05 and 0.05, as defined in the `IAAO Standard on Ratio Studies`_
+    Section 9.2.7. Higher PRB values indicate progressivity in assessment,
+    while negative values indicate regressivity.
+
+    .. _IAAO Standard on Ratio Studies: https://www.iaao.org/media/standards/Standard_on_Ratio_Studies.pdf
+
+    .. note: PRB is significantly less sensitive to outliers than PRD or COD.
+
+    :param estimate:
+        A list or ``pd.Series`` of estimated values.
+        Must be the same length as ``sale_price``.
+    :param sale_price:
+        A list or ``pd.Series`` of sale prices.
+        Must be the same length as ``estimate``.
+    :type estimate: Array-like numeric values
+    :type sale_price: Array-like numeric values
+
+    :return: A single float value containing the PRB of the inputs.
+    :rtype: float
+
+    :Example:
+
+    .. code-block:: python
+
+        # Calculate PRB:
+        import assesspy as ap
+
+        ap.prb(ap.ccao_sample().estimate, ap.ccao_sample().sale_price)
+    """
+    check_inputs(estimate, sale_price)
+    estimate = pd.Series(estimate, dtype=float)
+    sale_price = pd.Series(sale_price, dtype=float)
+    ratio: pd.Series = estimate / sale_price
+    median_ratio: float = ratio.median()
+
+    lhs: pd.Series = (ratio - median_ratio) / median_ratio
+    rhs: pd.Series = ((estimate / median_ratio) + sale_price).apply(
+        lambda x: math.log2(x / 2)
+    )
+
+    prb_model = sm.OLS(lhs.to_numpy(), rhs.to_numpy()).fit()
+    prb = float(prb_model.params[0])
+
+    return prb
+
+
+def _calculate_gini(
+    estimate: Union[list[int], list[float], pd.Series],
+    sale_price: Union[list[int], list[float], pd.Series],
+) -> tuple[float, float]:
+    """
+    Helper function to calculate the Gini coefficients of sales and estimated
+    values. Note that the estimated value Gini is based on the sale price order.
+    """
+    check_inputs(estimate, sale_price)
+
+    estimate = (
+        pd.Series(estimate, dtype=float)
+        .rename("estimate")
+        .reset_index(drop=True)
+    )
+    sale_price = (
+        pd.Series(sale_price, dtype=float)
+        .rename("sale_price")
+        .reset_index(drop=True)
+    )
+    df = pd.concat([estimate, sale_price], axis=1)
+    # Mergesort is required for stable sort results
+    df.sort_values(by="sale_price", kind="mergesort", inplace=True)
+    df.reset_index(drop=True, inplace=True)
```

Comment:
The index manipulation here is annoyingly important. If you pass two Series as inputs to this function and they have different indexes, the result will be a NaN output (due to index joining) unless you reset the indices.